### DISCUSSION TITLE
IS IT IMPOSSIBLE: Fetch only rows where there is no foreign relation to another table in React

### DISCUSSION THREAD

**Discussion by LavaGolem** [supabase/supabase#13798] [Questions] [answered]
Hi ppl, so I'm trying to fetch only rows of a table where no related foreign table relations match the requirements. I have created an SQL query that is working as expected, but I would like to do it on the frontend using Supabase Javascript SDK

Here is the SQL I want to mimic

```
SELECT *
FROM main_table
  AND NOT EXISTS (
    SELECT 1
    FROM foreign_connecting_table
    INNER JOIN foreign_table
      ON foreign_connecting_table.f_id = foreign_table.f_id
        AND foreign_table.is_current_version = true
    WHERE foreign_connecting_table.m_id = main_table.m_id
  )
LIMIT 1000;
```
What happens when I query with Supabase SDK is that I get ALL the rows, and just foreign table elements are empty, for ALL the rows, not just the ones where the connection doesn't exist

Here is my request: 

```
    const request = supabase
      .from('main_table')
      .select(
        `
        *,
        foreign_table(*),
      `,
        { count: 'exact' },
      )
```      

Thanks in Advance

**Comment by steve-chavez** [comment, marked as answer]
Hey @LavaGolem,

> Here is the SQL I want to mimic

For complex queries you can always resort to functions. You can wrap your query into a function like:

```sql
create or replace function main_tbl()
returns setof main_table as $$
  SELECT *
  FROM main_table
    AND NOT EXISTS (
      SELECT 1
      FROM foreign_connecting_table
      INNER JOIN foreign_table
        ON foreign_connecting_table.f_id = foreign_table.f_id
          AND foreign_table.is_current_version = true
      WHERE foreign_connecting_table.m_id = main_table.m_id
    )
  LIMIT 1000;
$$ language SQL
```

And call it in js through [rpc](https://supabase.com/docs/reference/javascript/rpc). Note you can use all the js methods on your rpc as it will return a normal table.

**Reply by LavaGolem** [reply]
That was my fear, thanks :D
